pkg/networkserver/grpc_gsns.go

```diff
@@ -706,24 +647,24 @@ func (ns *NetworkServer) handleUplink(ctx context.Context, up *ttnpb.UplinkMessa
 					logger.WithField("cid", cmd.CID).WithError(err).Warn("Failed to process MAC command")
 				}
 			}
-			if stored.MACState.LoRaWANVersion.Compare(ttnpb.MAC_V1_1) < 0 {
-				stored.Session = ses
-				stored.PendingSession = nil
-			} else if stored.PendingSession != nil {
-				handleErr = true
-				return nil, nil, errNoRekey
+			if pending {
+				if stored.MACState.LoRaWANVersion.Compare(ttnpb.MAC_V1_1) < 0 {
+					stored.Session = stored.PendingSession
+					stored.PendingSession = nil
+					stored.EndDeviceIdentifiers.DevAddr = &pld.DevAddr
```

Comment:
This is because we assume that the device registry treats setting identifiers as implicit, right?

Reply:
The device changes the `DevAddr` at this point and hence we record this transition.
Registry only uses the `session.dev_addr` and `pending_session.dev_addr`

Reply:
Clients take the `dev_addr` from the identifiers, not from the session

Reply:
This is why we set it here.

Reply:
> Registry only uses the `session.dev_addr` and `pending_session.dev_addr`

I mean is that `ids.dev_addr` is implicit by the registry. That's fine.